common/crypto_utils/avalon_crypto_utils/signature.py

```diff
@@ -269,21 +269,20 @@ def generate_client_signature(
             input_json_params, session_key,
             session_iv, worker.encryption_key, data_key, data_iv)
 
-        if input_json_params["requesterNonce"] and \
-                is_valid_hex_str(input_json_params["requesterNonce"]):
-            nonce = crypto.string_to_byte_array(
-                input_json_params["requesterNonce"])
-        else:
-            # [NO_OF_BYTES] 16 BYTES for nonce.
-            # This is the recommendation by NIST to
-            # avoid collisions by the "Birthday Paradox".
-            nonce = crypto.random_bit_string(NO_OF_BYTES)
-
-        request_nonce_hash = crypto.compute_message_hash(nonce)
-        nonce_hash = (crypto.byte_array_to_base64(request_nonce_hash)).encode(
-            'UTF-8')
+        if "requesterNonce" in input_json_params:
+            if len(input_json_params["requesterNonce"]) == 0:
+                # [NO_OF_BYTES] 16 BYTES for nonce.
+                # This is the recommendation by NIST to
+                # avoid collisions by the "Birthday Paradox".
+                input_json_params["requesterNonce"] = secrets.token_hex(
+                    NO_OF_BYTES)
+            elif not is_valid_hex_str(input_json_params["requesterNonce"]):
+                raise Exception("Invalid data format for requesterNonce")
+
```

Comment:
requesterNonce is a mandatory field and if it's not present in input request
LOG error and return fail status

Reply:
generate_client_signature() is utility function not an API. It is doing too many things, It is validating parameters, updating input request with many parameters like requesterNonce, encryptedRequestHash, encryptedSessionKey, verifyingKey. Ideally should update only requesterSignature as function name indicates.

Reply:
Yeah. That's correct. We should clean this by doing some parameter validation outside the function. Could be done later I suppose.